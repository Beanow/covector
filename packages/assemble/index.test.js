const { assemble, mergeIntoConfig } = require("./index");

describe("assemble changes", () => {
  const testTextOne = `
---
"assemble1": patch
"assemble2": patch
---
    
This is a test.
`;
  const testTextTwo = `
---
"assemble1": minor
"assemble2": patch
---
    
This is a test.
`;
  const testTextThree = `
---
"assemble1": patch
"assemble2": major
---
    
This is a test.
`;
  const testTextFour = `
---
"assemble1": patch
"@namespaced/assemble2": patch
---
    
This is a test.
`;

  it("runs", () => {
    const assembled = assemble([
      testTextOne,
      testTextTwo,
      testTextThree,
      testTextFour,
    ]);
    expect(assembled).toMatchSnapshot();
  });
});

describe("merge config test", () => {
  const assembledChanges = {
    releases: {
      "@namespaced/assemble2": {
        changes: [
          {
            releases: {
              "@namespaced/assemble2": "patch",
              assemble1: "patch",
            },
            summary: "This is a test.",
          },
        ],
        type: "patch",
      },
      assemble1: {
        changes: [
          {
            releases: {
              assemble1: "patch",
              assemble2: "patch",
            },
            summary: "This is a test.",
          },
          {
            releases: {
              assemble1: "minor",
              assemble2: "patch",
            },
            summary: "This is a test.",
          },
          {
            releases: {
              assemble1: "patch",
              assemble2: "major",
            },
            summary: "This is a test.",
          },
          {
            releases: {
              "@namespaced/assemble2": "patch",
              assemble1: "patch",
            },
            summary: "This is a test.",
          },
        ],
        type: "minor",
      },
      assemble2: {
        changes: [
          {
            releases: {
              assemble1: "patch",
              assemble2: "patch",
            },
            summary: "This is a test.",
          },
          {
            releases: {
              assemble1: "minor",
              assemble2: "patch",
            },
            summary: "This is a test.",
          },
          {
            releases: {
              assemble1: "patch",
              assemble2: "major",
            },
            summary: "This is a test.",
          },
        ],
        type: "major",
      },
    },
  };

  const config = {
    pkgManagers: {
      javascript: {
        version:
          "lerna version ${ release.type } --no-git-tag-version --no-push",
        publish: "npm publish",
      },
    },
    packages: {
      assemble1: {
        path: "./packages/assemble1",
        manager: "javascript",
      },
      assemble2: {
        path: "./packages/assemble2",
        version: "lerna version ${ release.type }",
      },
      "@namespaced/assemble1": {
        path: "./packages/namespaced-assemble2",
        manager: "cargo",
        version: "cargo version ${ release.type }",
        publish: "cargo publish",
      },
      "@namespaced/assemble2": {
        path: "./packages/namespaced-assemble2",
        manager: "cargo",
        version: "cargo version ${ release.type }",
        publish: "cargo publish",
      },
    },
  };

  it("merges version", () => {
    const mergedVersionConfig = mergeIntoConfig({
      config,
      assembledChanges,
      command: "version",
    });
    expect(mergedVersionConfig).toMatchSnapshot();
  });

  it("merges publish", () => {
    const mergedPublishConfig = mergeIntoConfig({
      config,
      assembledChanges,
      command: "publish",
    });
    expect(mergedPublishConfig).toMatchSnapshot();
  });
});
