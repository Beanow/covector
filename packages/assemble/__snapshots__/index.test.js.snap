// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assemble changes runs 1`] = `
Object {
  "changes": Array [
    Object {
      "releases": Object {
        "assemble1": "patch",
        "assemble2": "patch",
      },
      "summary": "This is a test.",
    },
    Object {
      "releases": Object {
        "assemble1": "minor",
        "assemble2": "patch",
      },
      "summary": "This is a test.",
    },
    Object {
      "releases": Object {
        "assemble1": "patch",
        "assemble2": "major",
      },
      "summary": "This is a test.",
    },
    Object {
      "releases": Object {
        "@namespaced/assemble2": "patch",
        "assemble1": "patch",
      },
      "summary": "This is a test.",
    },
  ],
  "releases": Object {
    "@namespaced/assemble2": Object {
      "changes": Array [
        Object {
          "releases": Object {
            "@namespaced/assemble2": "patch",
            "assemble1": "patch",
          },
          "summary": "This is a test.",
        },
      ],
      "type": "patch",
    },
    "assemble1": Object {
      "changes": Array [
        Object {
          "releases": Object {
            "assemble1": "patch",
            "assemble2": "patch",
          },
          "summary": "This is a test.",
        },
        Object {
          "releases": Object {
            "assemble1": "minor",
            "assemble2": "patch",
          },
          "summary": "This is a test.",
        },
        Object {
          "releases": Object {
            "assemble1": "patch",
            "assemble2": "major",
          },
          "summary": "This is a test.",
        },
        Object {
          "releases": Object {
            "@namespaced/assemble2": "patch",
            "assemble1": "patch",
          },
          "summary": "This is a test.",
        },
      ],
      "type": "minor",
    },
    "assemble2": Object {
      "changes": Array [
        Object {
          "releases": Object {
            "assemble1": "patch",
            "assemble2": "patch",
          },
          "summary": "This is a test.",
        },
        Object {
          "releases": Object {
            "assemble1": "minor",
            "assemble2": "patch",
          },
          "summary": "This is a test.",
        },
        Object {
          "releases": Object {
            "assemble1": "patch",
            "assemble2": "major",
          },
          "summary": "This is a test.",
        },
      ],
      "type": "major",
    },
  },
}
`;

exports[`merge config test merges publish 1`] = `
Array [
  Object {
    "path": "./packages/namespaced-assemble2",
    "pkg": "@namespaced/assemble2",
    "publish": "cargo publish",
  },
  Object {
    "path": "./packages/assemble1",
    "pkg": "assemble1",
    "publish": "npm publish",
  },
  null,
]
`;

exports[`merge config test merges version 1`] = `
Array [
  Object {
    "path": "./packages/namespaced-assemble2",
    "pkg": "@namespaced/assemble2",
    "version": "cargo version patch",
  },
  Object {
    "path": "./packages/assemble1",
    "pkg": "assemble1",
    "version": "lerna version minor --no-git-tag-version --no-push",
  },
  Object {
    "path": "./packages/assemble2",
    "pkg": "assemble2",
    "version": "lerna version major",
  },
]
`;
